esignet-service/docker-entrypoint.sh

```
@@ -1,13 +1,41 @@
+DIR_NAME=$hsm_local_dir_env
+FILE_NAME="client.zip"
+
+if [[ -n "$hsm_client_zip_url_env" ]]; then
+  has_parent=$(zipinfo -1 "$FILE_NAME" | awk '{split($NF,a,"/");print a[1]}' | sort -u | wc -l)
+  if test "$has_parent" -eq 1; then
+    echo "Zip has a parent directory inside"
+    dirname=$(zipinfo -1 "$FILE_NAME" | awk '{split($NF,a,"/");print a[1]}' | sort -u | head -n 1)
+    echo "Unzip directory"
+    unzip $FILE_NAME
+    echo "Renaming directory"
+    mv -v $dirname $DIR_NAME
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)